### PR TITLE
EPMEDU-728: SignIn page

### PR DESCRIPTION
**Pull request by DoTheMonkeyBusiness**
- created SignIn screen
- created Splash screen
- added navigation module
- added extensions module
- added foundation module
- added common module
- added resources module
- moved all resources into resources module

image

https://user-images.githubusercontent.com/36794185/177166782-563e13d4-420b-4631-b428-9b4aca553bc9.mov

### CODE REVIEW

File: library/resources/src/main/res/drawable-anydpi/ic_phone.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<vector android:height="24dp" android:viewportHeight="22"
+    android:viewportWidth="16" android:width="17.454546dp" xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (egorikftp) [inline library/resources/src/main/res/drawable-anydpi/ic_phone.xml]: could you please align code?

File: library/resources/src/main/res/drawable-anydpi/ic_facebook.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<vector android:height="24dp" android:viewportHeight="20"
+    android:viewportWidth="11" android:width="13.2dp" xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (egorikftp) [inline library/resources/src/main/res/drawable-anydpi/ic_facebook.xml]: could you please align code?

File: library/resources/src/main/res/drawable-anydpi/ic_animeal_logo.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:viewportHeight="110"
+    android:viewportWidth="110" android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (egorikftp) [inline library/resources/src/main/res/drawable-anydpi/ic_animeal_logo.xml]: the same

File: library/navigation/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+plugins {
+    id("AnimealPlugin")
+    id("com.android.library")
+}
+
+dependencies {
+    implementation(projects.base)
```
Comment (egorikftp) [inline library/navigation/build.gradle.kts]: do we need this in navigation?

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/TopBar.kt [hunk 5]
```diff
@@ -68,9 +56,10 @@ fun BackButton(onClick: () -> Unit) {
     }
 }
 
+@Suppress("UnusedPrivateMember")
```
Comment (egorikftp) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/TopBar.kt]: lets try this one

https://detekt.dev/docs/introduction/suppressors/

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.Dp
+
+@Composable
+fun HSpacer(
+    height: Dp,
```
Comment (egorikftp) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt]: can be in one line

File: library/resources/src/main/res/drawable-anydpi/ic_phone.xml (hunk 1 above)
Comment (DoTheMonkeyBusiness) [reply library/resources/src/main/res/drawable-anydpi/ic_phone.xml]: done

File: library/resources/src/main/res/drawable-anydpi/ic_facebook.xml (hunk 2 above)
Comment (DoTheMonkeyBusiness) [reply library/resources/src/main/res/drawable-anydpi/ic_facebook.xml]: done

File: library/resources/src/main/res/drawable-anydpi/ic_animeal_logo.xml (hunk 3 above)
Comment (DoTheMonkeyBusiness) [reply library/resources/src/main/res/drawable-anydpi/ic_animeal_logo.xml]: done

File: library/navigation/build.gradle.kts (hunk 4 above)
Comment (DoTheMonkeyBusiness) [reply library/navigation/build.gradle.kts]: no, we don't) removed

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/TopBar.kt (hunk 5 above)
Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/TopBar.kt]: It works nice, removed all _UnusedPrivateMember_ annotations from previews.

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt (hunk 6 above)
Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt]: done

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.Dp
+
+@Composable
+fun HSpacer(height: Dp) {
```
Comment (VEZE) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt]: it is worth avoiding abbreviations, perhaps it should have been called HeightSpacer or for ease of use and autocompletion SpacerHeight

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    Spacer(modifier = Modifier.height(height))
+}
+
+@Composable
+fun WSpacer(width: Dp) {
```
Comment (VEZE) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt]: it is worth avoiding abbreviations, perhaps it should have been called WidthSpacer or for ease of use and autocompletion SpacerWidth

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealButton.kt [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+fun AnimealButton(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+    color: Color = MaterialTheme.colors.primary,
+    enabled: Boolean = true,
+    content: @Composable RowScope.() -> Unit,
+) {
+    Button(
+        modifier = modifier
+            .height(56.dp)
```
Comment (VEZE) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealButton.kt]: Add name to call argument

File: feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/TabsScreen.kt [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        },
+    ) { padding ->
+        NavigationTabs(navigationController, padding)
+    }
+}
+
+@Composable
+private fun NavigationTabs(navigationController: NavHostController, padding: PaddingValues) {
+    ScreenNavHost(
+        modifier = Modifier.padding(padding),
```
Comment (VEZE) [inline feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/TabsScreen.kt:59]: Add name to call argument

File: feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/TabsScreen.kt [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        NavigationTab.Search,
+        NavigationTab.Favorites,
+        NavigationTab.Home,
+        NavigationTab.Analytics,
+        NavigationTab.More
+    )
+    BottomAppBar(backgroundColor = MaterialTheme.colors.surface) {
+        items.forEach { item ->
+            if (item == NavigationTab.Home) {
+                Box(modifier = Modifier.weight(1f))
```
Comment (VEZE) [inline feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/TabsScreen.kt:86]: Add name to call argument

File: feature/more/src/main/java/com/epmedu/animeal/more/root/ui/MoreOption.kt [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.epmedu.animeal.base.theme.AnimealTheme
+
+@Composable
+internal fun MoreOption(title: String, onClick: () -> Unit) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(40.dp)
```
Comment (VEZE) [inline feature/more/src/main/java/com/epmedu/animeal/more/root/ui/MoreOption.kt:22]: Add name to call argument

File: feature/more/src/main/java/com/epmedu/animeal/more/root/MoreScreenUi.kt [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    onNavigate: (String) -> Unit
+) {
+    Scaffold(
+        modifier = Modifier.fillMaxSize(),
+        topBar = { TopBar(title = stringResource(id = R.string.more)) }
+    ) { padding ->
+        Column(
+            modifier = Modifier.padding(padding)
+        ) {
+            Spacer(modifier = Modifier.height(48.dp))
```
Comment (VEZE) [inline feature/more/src/main/java/com/epmedu/animeal/more/root/MoreScreenUi.kt:31]: Add name to call argument

File: feature/login/src/main/java/com/epmedu/animeal/login/ui/LoginButtonContent.kt [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import com.epmedu.animeal.foundation.spacer.WSpacer
+import com.epmedu.animeal.resources.R
+
+@Composable
+internal fun RowScope.LoginButtonContent(
+    @DrawableRes iconId: Int,
+    @StringRes textId: Int,
+) {
+    Image(
+        modifier = Modifier.height(20.dp),
```
Comment (VEZE) [inline feature/login/src/main/java/com/epmedu/animeal/login/ui/LoginButtonContent.kt:27]: Add name to call argument

File: feature/login/src/main/java/com/epmedu/animeal/login/presentation/SignInScreenUI.kt [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+) {
+    Card(
+        modifier = modifier.fillMaxWidth(),
+        elevation = 20.dp,
+        shape = RoundedCornerShape(topStart = 24.dp, topEnd = 24.dp)
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(24.dp),
```
Comment (VEZE) [inline feature/login/src/main/java/com/epmedu/animeal/login/presentation/SignInScreenUI.kt:136]: Add name to call argument

File: base/src/main/java/com/epmedu/animeal/base/theme/Color.kt [hunk 16]
```diff
@@ -2,11 +2,23 @@ package com.epmedu.animeal.base.theme
 
 import androidx.compose.ui.graphics.Color
 
+val DisabledButtonColor = Color(0xFFD9DEE1)
```
Comment (VEZE) [inline base/src/main/java/com/epmedu/animeal/base/theme/Color.kt:5]: Should we add component variables to the color palette?

**Review by VEZE — APPROVED** [review]

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+    Spacer(modifier = Modifier.width(width))
+}
```
Comment (Fileek) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt:18]: Please rename them to HeightSpacer and WidthSpacer

**Review by Fileek — COMMENTED** [review]
Well done!

File: base/src/main/java/com/epmedu/animeal/base/theme/Color.kt (hunk 16 above)
Comment (Fileek) [reply base/src/main/java/com/epmedu/animeal/base/theme/Color.kt:5]: > Should we add component variables to the color palette?

That are my variables, they should be probably in some another place, but be available for all UI components

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt (hunk 17 above)
Comment (Fileek) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt:18]: Or SpacerHeight and SpacerWidth as @VEZE suggested

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt (hunk 7 above)
Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt]: Done, renamed to HeightSpacer.

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt (hunk 8 above)
Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt]: Done, renamed to WidthSpacer

File: base/src/main/java/com/epmedu/animeal/base/theme/Color.kt (hunk 16 above)
Comment (DoTheMonkeyBusiness) [reply base/src/main/java/com/epmedu/animeal/base/theme/Color.kt:5]: Will be done in scope of another ticket.

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt (hunk 17 above)
Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/spacer/Spacer.kt:18]: Done, renamed to WidthSpacer.

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealButton.kt (hunk 9 above)
Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealButton.kt]: I suggest not to add named arguments in places where there is only one argument and it is quite clear what it does.
cc: @egorikftp

Comment (DoTheMonkeyBusiness) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealButton.kt]: Agreed not to add in simple cases

**Review by egorikftp — APPROVED** [review]

**Review by Fileek — APPROVED** [review]

**Review by FlinForm — APPROVED** [review]

**Pull request merged**